### ISSUE TITLE
mask和激活函数位置交换导致的更新问题

### ISSUE THREAD

**Issue by Wangpeiyi9979** [thunlp/OpenNRE#139]
您好，贵组使用mask实现pcnn的想法是在让我佩服, 不过最近我用**pytorch**实现时遇到一个奇怪的问题：

`
 x = __piecewise_pooling__(x, mask) 

 x = activation(x)
`
**交换这两行代码位置**. 则会造成**梯度爆炸**的问题. 思考了很久也不知道是为什么。请问你们遇到过该问题吗. 万分感谢.

**Comment by gaotianyu1350** [comment]
感觉理论上应该不会有区别，tensorflow里面好像没这个问题？那大概是pytorch的bug......

以及pooling得是max pooling才行

**Comment by ShellingFord221** [comment]
你好，使用mask实现piece-wise pooling的代码看了不是很懂，可不可以大概讲下思路，或者有相关资料推荐下，非常感谢！！

**Comment by gaotianyu1350** [comment]
如果是不明白pytorch的操作的话，可以参考pytorch的[官方文档](https://pytorch.org/docs/stable/index.html)

本身的原理很简单，mask表示的是这个词属于三段中的具体哪一部分，算每一段pooling的时候，别的段落减去一个特别大的数字，这样max-pooling的时候一定不会选到他们

pcnn的论文可以参考[这篇](http://www.emnlp2015.org/proceedings/EMNLP/pdf/EMNLP203.pdf)
